tests/js/spec/components/autoplayVideo.spec.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import {render} from '@testing-library/react';
+
+import {AutoplayVideo} from 'sentry/components/autoplayVideo';
+
+jest.mock('react', () => {
+  return {
+    ...jest.requireActual('react'),
+    useRef: jest.fn(),
+  };
+});
+
+// Use a proxy to prevent <video ref={ref}/> from overriding our ref.current
+const makeProxyMock = (video: Partial<HTMLVideoElement>) => {
+  return new Proxy(
+    {current: video},
+    {
+      get(obj, prop) {
+        return obj[prop];
+      },
+      set(obj, prop) {
+        if (prop === 'current') {
+          obj.current = obj.current;
+        }
+        return true;
+      },
+    }
+  );
+};
+
+describe('autoplayVideo', () => {
+  it('sets mute and calls play', () => {
+    const mock = makeProxyMock({
+      muted: false,
+      play: jest.fn().mockReturnValue(Promise.resolve()),
+    });
+
+    // @ts-ignore we are mocking useRef
+    React.useRef.mockImplementation(() => mock);
+
+    const {container} = render(<AutoplayVideo src="https://example.com/video.mp4" />);
+
+    // Videos sadly do not have a role
+    // eslint-disable-next-line
+    expect(container.querySelector('video')).toBeInTheDocument();
+    expect(mock.current.muted).toBe(true);
+    expect(mock.current.play).toHaveBeenCalledTimes(1);
+  });
+
+  it('handles non promise-like return from play', () => {
+    const mock = makeProxyMock({
+      muted: false,
+      play: jest.fn().mockReturnValue(null),
+    });
+
+    // @ts-ignore we are mocking useRef
+    React.useRef.mockImplementation(() => mock);
+
+    const {container} = render(<AutoplayVideo src="https://example.com/video.mp4" />);
+
+    // Videos sadly do not have a role
```

Comment:
in this case, do you think a `data-test-id` would be better?

Reply:
Yup, let's do it. This was the first one I refactored, but let's do things more in rtl style indeed

Reply:
just out of curiosity I was checking why video does not have a role and found this

![image](https://user-images.githubusercontent.com/29228205/152112732-d1334805-a1df-44b7-bb40-1ab8c8af0f36.png)


so maybe we could use the role `application`, but in case not, I think an `aria-label `with a description would be better than a `data-test-id` .sorry for that

Reply:
100%, that's good curiosity :)!